### ISSUE TITLE
`UndefRefError` when accessing field `declaredType` in modelDescription

### ISSUE THREAD

**Issue by dafred94** [ThummeTo/FMI.jl#151]
```
using FMIZoo
using FMI

bball = fmiLoad(get_model_filename(1, "Dymola", "2022x"))

# now for the error...
bball.modelDescription.modelVariables[2]._Real.declaredType
```

Note that this only happens for specific `modelVariables`!

**Comment by dafred94** [comment]
Found a hint in https://github.com/ThummeTo/FMICore.jl/blob/d9898e702f8feeb383b080fe648bf80833948edc/src/FMI2_c.jl#L154.

The inner constructor does not assign to all fields.

**Comment by JoKircher** [comment]
This PR will resolve the issue, supporting declaredTypes correctly.

https://github.com/ThummeTo/FMIImport.jl/pull/79

**Comment by ThummeTo** [comment]
This is solved in v0.12.0
